### ISSUE TITLE
change image filename but not the extension

### ISSUE THREAD

**Issue by H889761123** [zofe/rapyd-admin#136]
Hi,
I am using this code to save an image

```
$edit->add('image','Image','image')->image(function($image){
    $name = $image->filename.time();
    $extension = $image->extension;
    $image->widen('800',function($constraint){$constraint->upsize();});
    $image->save("public/uploads/{$name}_width800.{$extension}");
}
```

The problem, is that when saving the image, the database field value is the orginal filename, I know I can use ->move() and pass the filename as parameter, but I don't know the extension of the original filename beforehand.

in summary, I want to change the filename value when saving it to database, but I want to keep the original uploaded file extension.

any ideas ?

**Comment by mehrdd** [comment]
+1

**Comment by zofe** [comment]
in summary, I want to change the filename value when saving it to database, 
    but I want to keep the   original uploaded file extension.
That seems reasonable

untested... but this can work, please let me know
```php
$edit->add('image','Image','image')->image(function($image) use ($edit){
    ...
    $edit->set('image',"{$name}_width800.{$extension}");
}
```

**Comment by mehrdd** [comment]
Thanks for the reply.
i tested your code but it didn't work ...

i used a dirty way for a demo like this :
```php
if (\Request::input('process'))
{
	$filename = \Request::file('image')->getClientOriginalName();
	$ext = \Request::file('image')->getClientOriginalExtension();
	$without_ext = preg_replace('/\\.[^.\\s]{3,4}$/', '', $filename);
	$filename = str_slug(strtolower($without_ext)).'-'.time().'.'.$ext;
}

$form->add('image', 'Category Image', 'image')
	->move(public_path()."/files/images", $filename)
	->rule('mimes:jpeg,jpg,png,gif')
	->preview(100, 100);
```

**Comment by zofe** [comment]
@mehrdd the sample I made is about image() closure  so it's more related to @HassanKanj  code,
However  we can adopt a convention like this..:  if given $filename  is without extension,  the original extension  will be added. This would be enough?

**Comment by mehrdd** [comment]
Yeah absolutely , The image() closure is very cleaner and better ...

**Comment by HassanKanj** [comment]
I have added the code, but unfortunately it didn't work, the code is still saving the original filename in the database

**Comment by HassanKanj** [comment]
I am still trying to solve this issue but with no success, any specific reason why this didn't work? or any alternative solution?

**Comment by fantasmewfs** [comment]
I just found an alternative soulution:
    
if(Input::hasFile('image')){
            $extension = Input::file('image')->getClientOriginalExtension();
        }else{
            $extension=NULL;
        }
...
 $entity->add('image','Image', 'image')->move($img_path, 'cool_name.'.$extension);
    
Have fun!

**Comment by dannoso** [comment]
In case somebody still need this i found also this to work:

```
        $form->saved(function () use ($form) {
            $image_info = pathinfo($form->model->photo);
            $form->model->setAttribute('photo', 'beehives/beehive_' . $form->model->id . '_' . time() . '.' . $image_info['extension']);
            $form->model->save();

            return redirect('panel/Beehive/all')->with('message', 'Record saved')->send();
        });
```

**Comment by heimasi** [comment]
In File.php, add codes behind line 108:

if (strstr($filename, '.') == false)
    $filename = $filename .'.'. strtolower($this->file->getClientOriginalExtension());

then, use move('path/to/', 'name_without_ext');

**Comment by laratutodotcom** [comment]
How to get file extension from path in Laravel 9? https://laratuto.com/how-to-get-file-extension-from-path-in-laravel/**

**```
$_PATH_INFO = pathinfo(public_path('/uploads/image.png')); 

$_EXTENSTION = $_PATH_INFO['extension'];

dd($_EXTENSTION );
```**

**Comment by laratutodotcom** [comment]
```
$_PATH_INFO = pathinfo(public_path('/uploads/image.png'));


$_EXTENSTION = $_PATH_INFO['extension'];

dd($_EXTENSTION );``
```
